iroh-rpc-client/src/macros.rs

```diff
@@ -2,88 +2,79 @@
 macro_rules! impl_client {
     ($label:ident) => {
         paste::paste! {
-            /// Name that the health service registers the client,
-            /// as this is derived from the protobuf definition.
-            #[cfg(feature = "grpc")]
-            pub(crate) const SERVICE_NAME: &str = stringify!([<$label:lower>].[<$label>]);
-
-            /// The display name that we expect to see in the StatusTable.
-            #[cfg(feature = "grpc")]
-            pub(crate) const NAME: &str = stringify!([<$label:lower>]);
-
             #[derive(Debug, Clone)]
             pub struct [<$label Client>] {
-                backend: [<$label ClientBackend>],
+                backend: std::sync::Arc<tokio::sync::RwLock<[<$label BackendState>]>>,
```

Comment:
This might give some contention.

Reply:
I haven't found a different way to get lazy connection working, which we need. It should be fine though, as the fast path only ever takes a read lock, so once connected, there should be no contention